modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/connector/ConnectionInterceptor.java

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.runtime.connector;
+
+import static org.mule.api.lifecycle.LifecycleUtils.disposeIfNeeded;
+import static org.mule.api.lifecycle.LifecycleUtils.initialiseIfNeeded;
+import static org.mule.api.lifecycle.LifecycleUtils.startIfNeeded;
+import static org.mule.api.lifecycle.LifecycleUtils.stopIfNeeded;
+import static org.mule.module.extension.internal.ExtensionProperties.CONNECTION_PARAM;
+import static org.mule.util.Preconditions.checkArgument;
+import org.mule.api.MuleContext;
+import org.mule.api.MuleException;
+import org.mule.api.lifecycle.InitialisationException;
+import org.mule.api.lifecycle.Lifecycle;
+import org.mule.extension.connection.ConnectionHandler;
+import org.mule.extension.runtime.Interceptor;
+import org.mule.extension.runtime.OperationContext;
+import org.mule.module.extension.internal.ExtensionProperties;
+import org.mule.module.extension.internal.runtime.OperationContextAdapter;
+
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implements simple connection management by using the {@link #before(OperationContext)} phase to
+ * set a connection as parameter value of key {@link ExtensionProperties#CONNECTION_PARAM} into an
+ * {@link OperationContext}.
+ * <p/>
+ * The connection is not actually established until the first invocation to the {@link #before(OperationContext)}
```

Comment:
This behaves assume a lot about the reusability and thread safety of the connection, rather than delegating these concerns to a connectionFactory/connectionHandler.